docs/upgrading-to-v0.18.x.md

```diff
@@ -0,0 +1,274 @@
+1. Create resources using the above manifests
+2. Verify that resources are `READY` and `SYNCED`
+
+#### Importing as Beta
+
+**Template to import `Cluster`:**
+
+```
+apiVersion: container.beta.gcp.crossplane.io/v1beta1
+kind: Cluster
+metadata:
+  name: <to-be-filled>
+  annotations:
+    crossplane.io/external-name: <to-be-filled>
+  labels:
+    <to-be-filled>
+spec:
+  # Consider setting deletionPolicy as "Orphan" to make sure you don't delete
+  # the cloud resource until you successfully validate the import
+  # deletionPolicy: Orphan
+  forProvider:
+    location: <to-be-filled>
+  providerConfigRef:
+    name: <to-be-filled>
+```
+
+**Template to import `NodePool`:**
+
+```
+apiVersion: container.beta.gcp.crossplane.io/v1alpha1
+kind: NodePool
+metadata:
+  name: <to-be-filled>
+  annotations:
+    crossplane.io/external-name: <to-be-filled>
+  labels:
+    <to-be-filled>
+spec:
+  # Consider setting deletionPolicy as "Orphan" to make sure you don't delete
+  # the cloud resource until you successfully validate the import
+  # deletionPolicy: Orphan
+  forProvider:
+    cluster: <to-be-filled>
+  providerConfigRef:
+    name: <to-be-filled>
+```
+
+##### Steps
+
+1. Deploy [provider-gcp-beta] on the cluster.
+2. Create `ProviderConfig` resource for [provider-gcp-beta].
+3. Create resources using the above manifests.
+4. Verify that resources are `READY` and `SYNCED`.
+
+### Updating compositions
+
+If you have compositions relying on the deprecated types, you would need to
+update them as well to use the new types with proper fields and configuration
+depending on whether you have chosen `beta` or `stable`. Until you switch
+using the new types, your composite resources would fail since the old types no
+longer exist.
+
+To prevent the updates in the compositions to trigger creation of new composed
+resources, please follow the steps below:
+
+1. Stop the Crossplane controllers by scaling down Crossplane deployment to 0
+   replicas.
+
+   ```
+   kubectl -n crossplane-system scale deployment crossplane --replicas=0
+   ```
+
+2. Update your compositions to use new types and configurations. If you have
+   installed via a configuration package, update package to the new version
+   with changes.
+
+3. Update `resourceRefs` in the spec of your composite resources to point
+   the imported resources, for example:
+
+   ```
+   spec:
+     compositionRef:
+       name: gke.gcp.platformref.crossplane.io
+     compositionUpdatePolicy: Automatic
+     id: platform-ref-gcp-cluster
+     parameters:
+       networkRef:
+         id: platform-ref-gcp-network
+       nodes:
+         count: 3
+         size: small
+     resourceRefs:
+     - apiVersion: container.gcp.crossplane.io/v1beta2 # <--- make sure the apiVersion matches with which you've used during import (e.g. beta vs stable)
+       kind: Cluster # <--- make sure type is correct
+       name: platform-ref-gcp-cluster-mwx8t-5j9hv # <--- make sure this is the name of the imported cluster
```

Comment:
For whatever reason, composite resource does not use the name we set here, rather create a new resource with generateName `platform-ref-gcp-cluster-mwx8t-` creating a new one and overrides the name here.

Suspected that I need all labels during import and added to the templates, but didn't help. The next thing I suspect is setting `ownerReferences`  as well.

Interestingly, this does not happen for `NodePool`.

Reply:
This is because resources are composed with old compositions once more before the package manager updates compositions with the new configuration package, added a workaround (temporarily break reference to compositon) to deal with that as well in the latest commit.